src/transformers/models/greaselm/processing_greaselm.py

```diff
@@ -0,0 +1,267 @@
+# coding=utf-8
+# Copyright 2021 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processor class for GreaseLM
+"""
+from collections import UserDict
+from typing import Any, Dict, List, Optional, Union
+
+import torch
```

Comment:
```suggestion
import torch
```
Hmm I see the processor also depends on PyTorch?

Can't we leverage BatchEncoding here, such that when users pass `return_tensors="pt",` this automatically gets converted in the appropriate format?

Reply:
We could, and I wanted to, but edge_index and edge_type are lists of lists where items are tensors. And they are of unequal lengths. Hence I introduced a `BatchEncoding` like-class called `LMKGEncoding` to recursively go through these lists and move tensors to the device.